add controllers of pipeline and pipelinerun

wujiahao15 opened the pull request:
There is something wrong with starting controller for pipeline and pipelinerun.

Commit:
init commit

JohnNiang commented on pkg/api/devops/v2alpha1/pipeline_types.go:
```diff
@@ -0,0 +1,140 @@
+}
+
+func init() {
+	SchemeBuilder.Register(&Pipeline{}, &PipelineList{})
```
```go
apis.AddToSchemes = append(apis.AddToSchemes, AddToScheme)
```

wujiahao15 replied:
Alright, it works. Thank you very much!

Commit:
feat(cmd/controller): add a flag to choose pipeline backend (jenkins or tekton)

We can now choose the controllers of pipeline related CRDs (Pipeline and PipelineRun) by specifying
flag `--pipeline-backend` when running `go cmd/controller/main.go`. The valid value of this flag is
`jenkins` or `tekton`.

Commit:
fix(controllers/tekton): fix logger missing error in controllers of pipeline and pipelinrun

Commit:
fix(pkg/apis): add devops.v2alpha1 and tekton.vbeta1 schemes

Commit:
chore(config/samples): add sample yaml files of Pipeline and PipelineRun

Commit:
ci(config/crd): add pipeline and pipelinerun crd yaml files

pipelinerun related yaml files, such as cainjection, webhook, editor_role, and viewer_role, are in
the initial state

wujiahao15 commented:
The current running environment can be described as follows:
1. Set up a kubernetes cluster using `kk` with version `v1.19.8`
2. Install `tekton-pipelines` using `helm install tekton cdf/tekton-pipelines`
3. Set up controller using command `go run cmd/controller/main.go --pipeline-backend=tekton`
    * This will run the tekton controllers of pipeline and pipelinerun.
    * If `pipeline-backend` flag is not specified, its default value is `jenkins`
    * This flag implementation can be reviewed in this [commit](https://github.com/flamywhale/ks-devops/pull/1/commits/e3c76d8978004aa63b643a64ee3bc16d6d0944bc).
4. Use `kubectl` to create Pipeline and PipelineRun to create a pipeline and run it.
    * Sample yaml files can be seen in this [commit](https://github.com/flamywhale/ks-devops/pull/1/commits/c6c155312d6e6f9e1f2e3e782d43df00602d0593).
